states/statefile/version4.go

```diff
@@ -202,6 +205,20 @@ func prepareStateV4(sV4 *stateV4) (*File, tfdiags.Diagnostics) {
 					}
 					deps = append(deps, ref.Subject)
 				}
+				obj.DependsOn = deps
```

Comment:
Yeah, `depends_on` contains references to modules and values as well. We need to load those into the graph to find the absolute resources they will eventually connect.

Reply:
That makes sense... thanks.

As a compromise, can we document the old field as being deprecated and read-only and make sure that any time we update a state object we'll force the old one to `nil` and migrate everything into the new one? At least then that puts us on a path to removing the deprecated field, because (if I'm understanding correctly) everything would be migrated over into the new field by the next refresh walk.

(I suspect this might already be true, because IIRC the apply step constructs a new state object from scratch each time rather than modifying the old one in-place, but I wanted to mention it explicitly just because I couldn't see any tests here that _prove_ it to be true by.)